assets/js/googlesitekit/datastore/site/urls.js

```diff
@@ -0,0 +1,122 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { getLocale } from '../../../util';
+const { createRegistrySelector } = Data;
+
+export const selectors = {
+	/**
+	 * Gets an external Google URL that includes the user's locale.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state          Data store's state.
+	 * @param {Object} [args]         Optional arguments for the resulting URL.
+	 * @param {string} [args.website] Base URL hostname and schema.
+	 * @param {string} [args.path]    URL path to build complete URL with starting slash.
+	 * @param {Object} [args.query]   Object to append query to the URL.
+	 * @param {string} [args.hash]    Optional hash.
+	 * @param {string} [args.locale]  Optional locale.
+	 * @return {(string|null)} The URL containing the user's locale or `null` if path is not set.
+	 */
+	getGoogleLocaleAwareURL( state, args ) {
+		const {
+			website,
+			path,
+			query,
+			hash,
+			locale = getLocale(),
+		} = args || {};
+
+		if ( ! path ) {
+			return null;
+		}
+
+		const url = new URL( website );
+
+		url.pathname = path;
+		url.hash = hash || '';
+
+		const params = { ...query, hl: locale };
+		for ( const key in params ) {
+			url.searchParams.set( key, params[ key ] );
+		}
+
+		return url.toString();
+	},
+
+	/**
+	 * Gets an external help link that includes the user's locale.
+	 *
+	 * @since 1.24.0
+	 *
+	 * @param {Object} state        Data store's state.
+	 * @param {Object} [args]       Optional arguments for the resulting URL.
+	 * @param {string} [args.path]  URL path to build complete URL with starting slash.
+	 * @param {Object} [args.query] Object to append query to the URL.
+	 * @param {string} [args.hash]  Optional hash.
+	 * @return {(string|null)} The URL containing the user's locale or `null` if path is not set.
+	 */
+	getGoogleSupportURL: createRegistrySelector( ( select ) => ( state, args ) => {
+		return select( STORE_NAME ).getGoogleLocaleAwareURL( {
+			...args,
+			website: 'https://support.google.com',
+		} );
+	} ),
+
+	/**
+	 * Gets the Google privacy policy URL.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {string} The Google privacy policy URL.
+	 */
+	getGooglePrivacyPolicyURL: createRegistrySelector( ( select ) => () => {
+		return select( STORE_NAME ).getGoogleLocaleAwareURL( {
+			website: 'https://myaccount.google.com',
+			path: '/privacypolicy',
+			locale: ( getLocale() || '' ).substr( 0, 5 ),
```

Comment:
Why the `.substr( 0, 5 )` here?

A locale like `https://policies.google.com/privacy?hl=zh-hans` is valid, and any extra data will be disregarded/redirected by Google here. Examples:

https://policies.google.com/privacy?hl=en-USA redirects to https://policies.google.com/privacy?hl=en

But https://policies.google.com/privacy?hl=zh-hans and https://policies.google.com/privacy?hl=zh-cw are both valid. I don't think we should be modifying these locale strings at all, as it seems Google is able to recognise longer ones and load the correct content.

Removing any parts of the locale _after_ the first two segments makes sense (see #2105), but we shouldn't assume the second part of locale will always be two characters, because that isn't always how they work.

I guess it depends on the locale standard used by WordPress, which is where we get this data from, right? I can't actually find, [in the WP docs](https://developer.wordpress.org/reference/functions/get_locale/), what the format they use is 🤷🏻‍♂️ 

I guess this might be fine, but is there any harm to passing the results of `getLocale()` directly?

Reply:
@tofumatt @eugene-manuilov Agreed. Let's not rely on any specific character length, we should only consider the first two segments though (essentially no matter their length). Having a third segment isn't really standardized (see #2105), so I think it's safer to cut off any additional segments more than the first two.

Reply:
Ok, updated.